.github/workflows/publish.yaml

```diff
@@ -0,0 +1,44 @@
+name: Publish to pub.dev
+
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+
+# Publish using the reusable workflow from dart-lang
+jobs:
+  publish:
+    name: Publish to pub.dev
+    environment: pub.dev
+    permissions:
+      contents: read
+      id-token: write # This is required for requesting the JWT
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout repository
+      - uses: actions/checkout@v3
+      # Setup Dart SDK with JWT token
+      - uses: subosito/flutter-action@48cafc24713cca54bbe03cdc3a423187d413aafa
+        with:
+          flutter-version: '3.16.0'
+          channel: stable
+          cache: true
+      # Minimal package setup and dry run checks
+      - name: Install dependencies
+        run: flutter pub get
+      - uses: dart-lang/setup-dart@a57a6c04cf7d4840e88432aad6281d1e125f0d46 
+      - name: Dry run 
+        run: dart pub publish --dry-run
```

Comment:
Can you also add a  dry run on every pull request?

Reply:
> Can you also add a dry run on every pull request?

You mean to fix the issues?

Reply:
No just to confirm that the changes in the PR will work and can be published to pub.dev

Reply:
> No just to confirm that the changes in the PR will work and can be published to pub.dev

Can you check main.yaml ?

Reply:
LGTM now